### PR TITLE
doc: Add 1.122 release note

### PR DESCRIPTION
**Pull request by yuwenma**
### Change description

Fixes #

### Tests you have done

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: docs/releasenotes/release-1.122.md [hunk 1]
```diff
@@ -1,26 +1,23 @@
...
-TODO: list contributors with `git log v1.121.0... | grep Merge | grep from | awk '{print $6}' | cut -d '/' -f 1 | sort | uniq`
+** This version is not yet released; this document is gathering release notes for the future release **
 
 ## Resources promoted from alpha to beta:
 
-*When resources are promoted from alpha to beta, we (generally) ensure they follow our best practices: use of refs on fields where appropriate,
-output fields from GCP APIs are in `status.observedState.*`
-
-* `PlaceholderKind`
+* `RedisCluster`
```
Comment (justinsb) [inline docs/releasenotes/release-1.122.md:9]: I think this is a new alpha resource, rather than alpha -> beta

**Comment by justinsb** [comment]
I think RedisCluster is net-new, but otherwise lgtm

File: docs/releasenotes/release-1.122.md (hunk 1 above)
Comment (yuwenma) [reply docs/releasenotes/release-1.122.md:9]: Updated. Also added the RunJob and SQLInstance Clone change.

**Comment by yuwenma** [comment]
> I think RedisCluster is net-new, but otherwise lgtm

So I put `RedisCluster` in the "Direct Cloud Reconciler" section. But you are right, it worths calling out that this is a net new resource since people is looking forward to it!

**Pull request merged**